tests/acceptance/settings-page/billing-test.js

```diff
@@ -0,0 +1,152 @@
+import { module, test } from 'qunit';
+import { setupApplicationTest } from 'codecrafters-frontend/tests/helpers';
+import { signIn, signInAsSubscriber, signInAsVipUser } from 'codecrafters-frontend/tests/support/authentication-helpers';
+import testScenario from 'codecrafters-frontend/mirage/scenarios/test';
+import billingPage from 'codecrafters-frontend/tests/pages/settings/billing-page';
+import { setupMirage } from 'ember-cli-mirage/test-support';
+import { setupWindowMock } from 'ember-window-mock/test-support';
+import percySnapshot from '@percy/ember';
+
+module('Acceptance | settings-page | billing-test', function (hooks) {
+  setupApplicationTest(hooks);
+  setupMirage(hooks);
+  setupWindowMock(hooks);
+
+  test('membership section shows correct plan for subscriber with active subscription', async function (assert) {
+    testScenario(this.server);
+    signInAsSubscriber(this.owner, this.server);
+    const subscription = this.server.schema.subscriptions.first();
+    subscription.update('pricingPlanName', 'Yearly Plan');
+
+    await billingPage.visit();
+
+    assert.ok(billingPage.membershipSection.isVisible, 'membership section is visible');
+    assert.ok(billingPage.membershipSection.text.includes('Membership active'), 'shows active plan');
+
+    await percySnapshot('Billing Page - Active Subscription');
+  });
+
+  test('membership section shows correct plan for non-subscriber', async function (assert) {
+    testScenario(this.server);
+    signIn(this.owner, this.server);
+
+    await billingPage.visit();
+
+    assert.ok(billingPage.membershipSection.isVisible, 'membership section is visible');
+    assert.notOk(billingPage.membershipSection.text.includes('Membership active'), 'does not show active plan');
```

Comment:
```suggestion
    assert.ok(billingPage.membershipSection.text.includes('Membership not found'), 'does not show active plan');
```

More reliable to test for a positive behaviour in tests, just because "membership active" isn't present it doesn't mean that whatever is present is correct